### PR TITLE
MNIST - Save/Load model as state dict

### PR DESCRIPTION
**Pull request by shrinath-suresh**
Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

## What changes are proposed in this pull request?

Updating the MNIST example to save the model as state dict at the end of the training process. 

This example illustrates on how to deploy a model in multi gpu environment. 

This PR is dependent on - https://github.com/mlflow/mlflow/pull/3705

## How is this patch tested?

Tested with End to End flow

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow TorchServe Deployment Plugin users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?
Components
- [ ] `area/deploy`: Main deployment plugin logic
- [ ] `area/build`: Build and test infrastructure for MLflow TorchServe Deployment Plugin
- [ ] `area/docs`: MLflow TorchServe Deployment Plugin documentation pages
- [x] `area/examples`: Example code

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: examples/MNIST/README.md [hunk 1]
```diff
@@ -15,7 +15,8 @@ Run the following command to train MNIST model
 CPU: `mlflow run . -P max_epochs=5`
 GPU: `mlflow run . -P max_epochs=5 -P gpus=2 -P accelerator=ddp`
 
-On the training completion, the MNIST model is stored as "model.pth" in current working directory.
+At the end of the training, MNIST model will be logged into mlflow as state dict and registers the mnist
```
Comment (HamidShojanazeri) [inline examples/MNIST/README.md:18]: @shrinath-suresh  Please consider adding the proper steps of running a server with sql backend and setting the tracking uri, prior to running the script.

**Review by HamidShojanazeri — COMMENTED** [review]
@shrinath-suresh Please refer to inline comments.

File: examples/MNIST/README.md (hunk 1 above)
Comment (shrinath-suresh) [reply examples/MNIST/README.md:18]: makes sense. I will update the README.md file to have sql backend as prerequisite.

Comment (shrinath-suresh) [reply examples/MNIST/README.md:18]: @ankan94 Ankan do we have the rst file ready for sql backend ?

**Comment by chauhang** [comment]
Depends on https://github.com/mlflow/mlflow/pull/3705 to be merged first before this PR can be merged.

**Comment by shrinath-suresh** [comment]
Marking it as WIP until the design for https://github.com/mlflow/mlflow/pull/3705 is finalized

File: examples/MNIST/conda.yaml [hunk 2]
```diff
@@ -14,4 +14,4 @@ dependencies:
   - pytorch_lightning>=1.0.2
   - pandas
   - gorilla
-  - mlflow
+  - mlflow @ git+https://github.com/mlflow/mlflow@master
```
Comment (savagej) [inline examples/MNIST/conda.yaml:17]: This can be changed to version 1.14 now right?
```suggestion
  - mlflow>=1.14.0
```

File: setup.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
     # Require MLflow as a dependency of the plugin, so that plugin users can simply install
     # the plugin & then immediately use it with MLflow
     install_requires=[
-        "mlflow>=1.12.0",
+        "mlflow @ git+https://github.com/mlflow/mlflow@master",
```
Comment (savagej) [inline setup.py:14]:
```suggestion
        "mlflow>=1.14.0",
```

**Pull request merged**